Source/WebCore/page/History.cpp

```diff
@@ -58,11 +58,16 @@ History::History(LocalDOMWindow& window)
 {
 }
 
-unsigned History::length() const
+static Exception documentNotFullyActiveException()
+{
+    return Exception { ExceptionCode::SecurityError, "Attempt to use History API from a document that isn't fully active"_s };
```

Comment:
Can you leave out `Exception` here?

Reply:
```
return Exception { ExceptionCode::SecurityError, "Attempt to use History API from a document that isn't fully active"_s };
```
> Can you leave out `Exception` here?

Is there a way to do that while also setting the "_Attempt to use History API from a document that isn't fully active"_ custom message?

Or do you mean that custom message seems unnecessary, and it’s sufficient to just throw a `SecurityError` without any custom message?

Reply:
It might be to rename function to `documentNotFullyActive` without word 'Exception' because it is explicit from function type.

Reply:
> It might be to rename function to `documentNotFullyActive` without word 'Exception' because it is explicit from function type.

aha — thanks, yes, changed it now to just `documentNotFullyActive`

Reply:
I meant to ask a question about something else, much simpler. Can the return statement just be `return { … }` instead of `return Exception { … }`?

Reply:
The rename also seems fine.

Reply:
> I meant to ask a question about something else, much simpler. Can the return statement just be `return { … }` instead of `return Exception { … }`?

ah OK, I see now. So when I try that, I get a _“chosen constructor is explicit in copy-initialization”_ error from clang:
```
/opt/WebKit/Source/WebCore/page/History.cpp:68:12: error: chosen constructor is explicit in copy-initialization
    return { ExceptionCode::SecurityError, "Attempt to use History API from a document that isn't fully active"_s };
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
/opt/WebKit/Source/WebCore/dom/Exception.h:50:19: note: explicit constructor declared here
inline Exception::Exception(ExceptionCode code, String message)
                  ^
```
And looking at the source in `dom/Exception.h`, that seems to make sense — though the actual precise cause is the `explicit` keyword up at line 36:
https://github.com/WebKit/WebKit/blob/8a9ddac3f43f8b4bd95d3427a86612e216947c63/Source/WebCore/dom/Exception.h#L34-L36

So it seems the compiler doing what it’s been told to do, and would only allow `return { ExceptionCode::SecurityError, … }` if that constructor weren’t intentionally declared `explicit`?

Reply:
Yes that’s fine. So the answer is no, can’t leave it out.